src/decorators/restrict.js

```diff
@@ -0,0 +1,6 @@
+export default function restrict(func) {
+	return (target, key, descriptor) => {
```

Comment:
Look how needPermissions decorator works.
Maybe this will be a cleaner way to do it, without having to touch the `Migi.js` file.